### PR TITLE
Use files for retrieving the addresses.

### PR DESCRIPTION
**Pull request by andreasnoack**
When launching the slurm workers I redirect the worker information to files instead of `STDOUT`. The reason for this is that, when working with several nodes, the lines got messed up and were difficult to parse. Writing to files might also make it possible to reconnect, but that is another issue.

It is the first time I write code like this so there is probably things that can be improved, but with this, I'm able to launch 512 workers on the cluster.

We probably want to look into options for specifying `-N` and `-c` instead of only `-n` as the user probably would like to have some control over how the workers are distributed across nodes.

### CODE REVIEW

File: src/slurm.jl [hunk 1]
```diff
@@ -34,24 +34,58 @@ function launch(manager::SlurmManager, params::Dict, instances_arr::Array,
...
+            if length(w) > 0
+                config = WorkerConfig()
+                config.port = parse(Int, w[1])
+                config.host = strip(w[2])
+                # Keep a reference to the proc, so it's properly closed once
+                # the last worker exits.
+                config.userdata = srun_proc
+                push!(instances_arr, config)
+                notify(c)
+            end
```
Comment (amitmurthy) [inline src/slurm.jl]: You may get a little more speedup by wrapping individual output file reads in `@async` blocks, within an overall `@sync` block.

File: src/slurm.jl [hunk 2]
```diff
@@ -34,24 +34,58 @@ function launch(manager::SlurmManager, params::Dict, instances_arr::Array,
...
+                    break
+                end
+                sleep(0.001)
+                if isfile(fn)
+                    t0 = time()
+                    while true
+                        if time() > t0 + 60 + np
+                            warn("dropping worker: file created, but not written to in $(60 + np) seconds")
+                            break
+                        end
```
Comment (amitmurthy) [inline src/slurm.jl]: Can combine into a single `isfile(fn) && (filesize(fn) > 0)` check.

**Comment by andreasnoack** [comment]
Thanks for the comments. I've updated my branch with your suggested change to the test condition. The async block didn't affect the launch time.

I've also added a line to clean up old Slurm output files .

**Comment by amitmurthy** [comment]
OK. LGTM. Please go ahead and merge whenever you are ready.

**Comment by andreasnoack** [comment]
I'll do that. @jakebolewski is looking into the C API so eventually we might use that instead, but for now this is able to get the job done on the big cluster.

**Pull request merged**